src/languages/yaml.js

```diff
@@ -54,12 +54,12 @@ function(hljs) {
         relevance: 10
       },
       { // multi line string
+        // Blocks start with a | or > followed by a newline
+        //
+        // Indentation of subsequent lines must be the same to
+        // be considered part of the block
         className: 'string',
-        begin: '[\\|>] *$',
-        returnEnd: true,
-        contains: STRING.contains,
-        // very simple termination: next hash key
-        end: KEY.variants[0].begin
+        begin: '[\\|>][ ]*\\n( *)[\\S ]+\\n(\\1[\\S ]+\\n?)*',
```

Comment:
Is this matching the whole block now? Do we not need STRING.contains anymore?

Reply:
Yeah it's matching the whole block, as it was the only way I could find to enforce the consistent indentation rule.

Reply:
Yeah, right now the engine really has no other way to save state like that. :-/

Reply:
So does this mean we lose escaping and variables inside these strings?

Reply:
Sorry I guess I asked that earlier but is your answer mean you think that losing this is a reasonable trade-off?

Reply:
I'm not super confident here, as I find the YAML spec really hard to understand...but I believe that this would be the right way to treat these block literals. My understanding is that everything inside is just that block, and it shouldn't be marked up further...but I could be wrong.

Relevant part of the spec: https://yaml.org/spec/1.2/spec.html#Block

Reply:
wow, yeah.

Things that jumped out as pertinent to me:

> Inside literal scalars, all (indented) characters are considered to be content, including white space characters.

So in the literal form sounds like it's 100% content for sure.  And really there is nothing you could really even want to escape I suppose other than line breaks and whitespace.

Looking at the block form again now.

Reply:
After reading that I'm tempting to say we should add Indentation indicators (0-9) and chomping indicttors (+/-) just to parse them, but I'm fine if we don't. :-)

I don't see anything about escape sequences or anything... I think I was just hung up that before we were using STRING.contains.

Reply:
They were easy enough to add, so I've put them in there.

Yeah I don't know how correct it is to be having any `contains` things inside the YAML strings. This grammar looks to have been built to cater for a specific use-case (something with Ruby + Ansible syntax inside) so I haven't gone and changed any of that, but I also haven't treated it as a constraint for fixing the blocks.